logger/buffered_logger.go

```diff
@@ -337,28 +347,34 @@ func (b *logBuffer) Dequeue() (*msg, error) {
 		b.wait.Wait()
 	}
 
+	// If ring buffer is closed, return ring buffer is closed error to notify the
+	// buffered log that the ring buffer is closed.
+	if b.isClosed {
```

Comment:
We already makes sure that the log is only sent when the buffer is not closed. Why do we throw an error here if the buffer is closed?

Reply:
Good point, I will reconsider the logic here

Reply:
Removed the error returned here in new commit